app/services/search_service.rb

```diff
@@ -82,30 +82,28 @@ def search_questions(input, limit = 25, order = :natural, type = :boolean)
   end
 
   # Search nearby nodes with respect to given latitude, longitute and tags
-  def tagNearbyNodes(query, tag, limit = 10)
-    raise("Must separate coordinates with ,") unless query.include? ","
-
-    lat, lon =  query.split(',')
+  def tagNearbyNodes(coordinates, tag, limit = 10)
+    raise("Must contain all four coordinates") if coordinates["nwlat"].nil?
+    raise("Must contain all four coordinates") if coordinates["nwlng"].nil?
+    raise("Must contain all four coordinates") if coordinates["selat"].nil?
+    raise("Must contain all four coordinates") if coordinates["selng"].nil?
 
-    raise("Must have at least one digit after .") unless lat.include? "."
-    raise("Must have at least one digit after .") unless lon.include? "."
+    raise("Must be a float") unless coordinates["nwlat"].is_a? Float
+    raise("Must be a float") unless coordinates["nwlng"].is_a? Float
+    raise("Must be a float") unless coordinates["selat"].is_a? Float
+    raise("Must be a float") unless coordinates["selng"].is_a? Float
 
-    nodes_scope = NodeTag.joins(:tag)
-      .where('name LIKE ?', 'lat:' + lat[0..lat.length - 2] + '%')
+    nodes_scope = NodeTag.joins(:tag).where('name LIKE ?', 'lat%').where('REPLACE(name, "lat:", "") BETWEEN ' + coordinates["selat"].to_s + ' AND ' + coordinates["nwlat"].to_s)
```

Comment:
Hi ,
I think we can improve this logic . Think of this as 2 diagonally opposite points : 
x1 = nw.lat
y1 = nw.lon
x2 = se.lat
y2 = se.lon

Now we want all those points (x,y) such that : 
1.) x1 <= x <= x2  
2.) y2 <= y <= y1
Both condition should be satisfied .

What do you think ?

@jywarren ...kindly verify this logic .

Reply:
Also can we do this using one `where` statement ? Just trying to optimize this query 😄 . Thanks !

Reply:
First off, what logic would you like to improve? Because everything in my fork works just as you described it. First it takes all nodes between `selat` and `nwlat` in here.
```suggestion
nodes_scope = NodeTag.joins(:tag).where('name LIKE ?', 'lat%').where('REPLACE(name, "lat:", "") BETWEEN ' + coordinates["selat"].to_s + ' AND ' + coordinates["nwlat"].to_s)
```
And then from this nodes it takes all between `nwlng` and `selng`.
```suggestion
items = Node.includes(:tag).references(:node, :term_data).where('node.nid IN (?)', nids)
      .where('term_data.name LIKE ?', 'lon%').where('REPLACE(term_data.name, "lon:", "") BETWEEN ' + coordinates["nwlng"].to_s + ' AND ' + coordinates["selng"].to_s)
```
I don't know what's wrong in here, because it does just what you described.

Reply:
Secondly, these `.where()` statements in the code are multiple but the request is only one. It's just for the purpose of easier code reading.
Here is the DB request log from line of code.
`SELECT "community_tags".* FROM "community_tags" INNER JOIN "term_data" ON "term_data"."tid" = "community_tags"."tid" WHERE (name LIKE 'lat%') AND (REPLACE(name, "lat:", "") BETWEEN 0.0 AND 200.0)`

If there is anything I should change, please let me know. Thank you for your help.

Reply:
Wow this is really cool SQL! Very neat!